### PR TITLE
Redis metadata storage

### PR DESCRIPTION
**Pull request by galfert**

### CODE REVIEW

File: lib/remote_storage/swift.rb [hunk 1]
```diff
@@ -255,11 +264,30 @@ def parent_directories_for(directory)
       parent_directories
     end
 
+    def parent_directory_for(directory)
+      if directory.match(/\//)
+        return directory[0..directory.rindex("/")-1]
```
Comment (skddc) [inline lib/remote_storage/swift.rb]: I think this is very contrived. How about sth more like:

```rb
if directory.include?("/")
  return directory.match(/^(.*)\//)[1]
```

File: lib/remote_storage/swift.rb [hunk 2]
```diff
@@ -255,11 +264,30 @@ def parent_directories_for(directory)
...
+      elsif directory != ""
```
Comment (skddc) [inline lib/remote_storage/swift.rb]: What's this case?

File: spec/swift/app_spec.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+      it "creates the metadata object in redis" do
+        put_stub = OpenStruct.new(headers: {etag: "bla"})
+        RemoteStorage::Swift.stub_any_instance :has_name_collision?, false do
+          RestClient.stub :put, put_stub do
+            put "/phil/food/aguacate", "si"
+          end
+        end
+
+        metadata = redis.hgetall "rs_meta:phil:food/aguacate"
```
Comment (skddc) [inline spec/swift/app_spec.rb]: Looks like these are not deleted after test runs, so the test might actually work although writing failed.

**Comment by skddc** [comment]
What's up? We can't continue with this, because our last status is that you continued on it and didn't push anything. We wanted to deploy this today.

**Comment by skddc** [comment]
@galfert On IRC you said this is ready to test live, but specs are red here and it still has a WIP tag on it. Could you explain and lead the way forward?

**Comment by galfert** [comment]
Finally got the specs to be green on Travis. Although I haven't figured out a proper way to run both the Riak and Swift specs. It's only running the Swift specs for now.

Tomorrow morning I will deploy this to staging, see that everything still works with the legacy setting. Then I will finish up the migration script and run it on staging.

**Comment by skddc** [comment]
Great, thanks for the update!

**Comment by skddc** [comment]
... btw, I think we should delete the Riak option for now, or at least rewrite things so that it's just used for storing the documents, same as Swift after this.

**Pull request merged**